app/models/kubernetes.js

```diff
@@ -34,33 +35,62 @@ exports.Model = Model;
 
 
 class Deployment extends Model {
+  constructor(data) {
+    super(data);
+    this.pods = new base.ModelSet(Pod, []);
+  }
+
   static get endpoint() {
     return 'deployments';
   }
 
-  static restart(label) {
-    return Pod.delete_all({ labelSelector: `app=${label}` });
+  get_pods() {
+    if (!this.pods.length) {
+      return Pod.list({ labelSelector: `app=${this.app_label}` }).
+        then((pods) => {
+          this._pods = pods;
+          return pods;
+        });
+    }
+    return Promise.resolve(this.pods);
   }
 
-  get_pods() {
-    return Pod.list({ labelSelector: `app=${this.data.metadata.labels.app}` });
+  static list() {
+    return Promise.all([super.list(), Pod.list()])
+      .then(([deployments, pods]) => {
+        const app_label = x => x.metadata.labels.app;
+        const pod_groups = group_by(pods, app_label);
+
+        return deployments.map((deployment) => {
+          deployment.pods = new base.ModelSet(Pod, pod_groups[app_label(deployment)]);
+          return deployment;
+        });
+      });
+  }
+
+  static create(data) {
+    return api.post(`tools/${data.tool_name}/deployments`, {});
   }
 
   restart() {
     return Pod.delete_all({ labelSelector: `app=${this.data.metadata.labels.app}` });
   }
 
-  get_status() {
-    for (let i = 0; i < this.status.conditions.length; i += 1) {
-      const condition = this.status.conditions[i];
-      if (condition.type === 'Available') {
-        if (condition.status === 'True') {
-          return 'Available';
-        }
-        return 'Unavailable';
-      }
-    }
-    return 'Unknown';
+  get available() {
+    return this.data.pods.some(pod => pod.status.phase === 'Running');
+  }
+
+  get app_label() {
+    return this.data.metadata.labels.app;
+  }
+
+  get username() {
+    // might get a truncated username
+    return this.data.metadata.namespace.split('user-').join('');
```

Comment:
One minor thing is that if the username starts with `user-`, you may get a namespace like `user-user-42` and this would not work.

We probably don't have any case like that so don't worry too much I guess, it's just another of these edge cases to bear in mind (especially if it comes to security).

Reply:
That is a good point.